Pipedream source documentation (#1067)
rudderlabs/rudderstack-docs: docs/sources/event-streams/cloud-apps/pipedream.mdx

---
title: "Pipedream"
description: Ingesting your event data from Pipedream into RudderStack.
aliases: ["Pipedream source"]
---

[Pipedream](https://pipedream.com/) lets you build and automate processes that connect APIs. It supports open source triggers and actions for hundreds of integrations.

This guide will help you set up Pipedream as a source in RudderStack.

## Getting started

Follow these steps to set up your Pipedream source in the RudderStack dashboard:

1. Go to your [RudderStack dashboard](https://app.rudderstack.com/) and click **Add Source**. From the list of **Event Streams** sources, select **Pipedream**.
2. Assign a name to your source and click **Continue**.
3. Your Pipedream source is now configured. Note the **Webhook URL** in the **Settings** tab, as shown:

<img src="../../../assets/event-stream-sources/pipedream-webhook.png" alt="Pipedream source webhook URL" />

4. Log in to your [Pipedream dashboard](https://pipedream.com/workflows/) and set up your workflow.

> 5. After <a href="https://pipedream.com/new?tutorial=1">adding a trigger</a>, click the **+** icon to add a step. Then, select the **HTTP / Webhook** option and choose **Send any HTTP Request**, as shown:

Reviewer: Can we accommodate this statement as well. User can select any other HTTP type as well but rudderstack would take only POST requests made to Ruderstack.

Author: I've added this as a note below Step 6.

Reviewer: Yeah I saw but it is not only for this source it is in genral. So need to update Note according to it

<img src="../../../assets/event-stream-sources/pipedream-http-webhook.png" alt="Pipedream add a step" />

> 6. In the **HTTP Request Configuration** section, choose **POST** and enter the RudderStack webhook URL obtained in Step 3 above. Also, configure the other request configuration settings as required.

Author: ```suggestion
6. In the **HTTP Request Configuration** section, choose **POST** as the request type and enter the RudderStack webhook URL obtained in Step 3 above. Also, configure the other request configuration settings as required.
```

<img src="../../../assets/event-stream-sources/pipedream-custom-http-request.png" alt="Pipedream custom HTTP request" />

7. Click **Deploy** to finish setting up the workflow.

## How RudderStack creates the event payload

RudderStack supports all the <Link to="/event-spec/standard-events/">standard events</Link> in the specified format. Otherwise, it transforms the incoming data to a <Link to="/event-spec/standard-events/track/">`track`</Link> payload.

A sample incoming payload is as shown:

```json
{
    "artist": "Artist One",
    "genre": "Jazz",
    "song": "Take Five"
  }
```

RudderStack transforms the above payload into the following `track` payload: 

```json
{
  "event": "pipedream_soure_event",
  "anonymousId": "63767499ca6fb1b7c988d5bb",
  "integration": {
    "name": "PIPEDREAM"
  },
  "integrations": {
    "PIPEDREAM": false
  },
  "type": "track",
  "properties": {
    "artist": "Artist One",
    "genre": "Jazz",
    "song": "Take Five"
  }
}
```

<br />

### Event transformation

> RudderStack ingests the Pipedream events after converting them into the RudderStack event format. It also maps the following properties from the Pipedream event payload to the RudderStack properties:

Reviewer: ```suggestion
RudderStack ingests the Pipedream events and checks for userId or anonymousId and type (identify ,track etc.) . If both are present it sees it as a rudderstack format payload other wise converts it into rudderstack payload format according to the below mapping:
```

> | Pipedream property  | RudderStack property| 

Reviewer: ```suggestion
| RudderStack property  | Value | 
```

> | :------------------ | :------------- |
> | `pipedream_source_event`  | `message.event`    | 
> | `track`    | `message.type`        | 
> | input payload       | `message.properties`|

Reviewer: ```suggestion
| :------------------ | :------------- |
| `message.event`  | `pipedream_source_event`    | 
|  `message.type`   |  `track`       | 
| `message.properties`     | pipedream input payload |
```

---
title: "Pipedream"
description: Ingesting your event data from Pipedream into RudderStack.
aliases: ["Pipedream source"]
---

[Pipedream](https://pipedream.com/) lets you build and automate processes that connect APIs. It supports open source triggers and actions for hundreds of integrations.

> This guide will help you set up Pipedream as a source in RudderStack.

Reviewer 2: can skip if seems fine to you as well

Author: Agreed - made the change.

3. Your Pipedream source is now configured. Note the **Webhook URL** in the **Settings** tab, as shown:

<img src="../../../assets/event-stream-sources/pipedream-webhook.png" alt="Pipedream source webhook URL" />

4. Log in to your [Pipedream dashboard](https://pipedream.com/workflows/) and set up your workflow.
5. After <a href="https://pipedream.com/new?tutorial=1">adding a trigger</a>, click the **+** icon to add a step. Then, select the **HTTP / Webhook** option and choose **Send any HTTP Request**, as shown:

<img src="../../../assets/event-stream-sources/pipedream-http-webhook.png" alt="Pipedream add a step" />

6. In the **HTTP Request Configuration** section, choose **POST** as the request type and enter the RudderStack webhook URL obtained in Step 3 above. Also, configure the other request configuration settings as required.

<div class="warningBlock">
RudderStack only accepts the incoming HTTP requests of type <strong>POST</strong>.
</div>

<img src="../../../assets/event-stream-sources/pipedream-custom-http-request.png" alt="Pipedream custom HTTP request" />

7. Click **Deploy** to finish setting up the workflow.

## How RudderStack creates the event payload

> RudderStack supports all the <Link to="/event-spec/standard-events/">standard events</Link> in the specified format. Otherwise, it transforms the incoming data to a <Link to="/event-spec/standard-events/track/">`track`</Link> payload.

Reviewer 2: ```suggestion
RudderStack supports all the <Link to="/event-spec/standard-events/">standard events</Link> in the specified format. For the rest of the events, it transforms the incoming data to a <Link to="/event-spec/standard-events/track/">`track`</Link> payload.
```

Author: Slightly tweaked to "For the other events..." to make it a little less wordy. WDYT?

A sample incoming payload is as shown:

```json
{
    "artist": "Artist One",
    "genre": "Jazz",
    "song": "Take Five"
  }
```

RudderStack transforms the above payload into the following `track` payload: 

```json
{
  "event": "pipedream_soure_event",
  "anonymousId": "63767499ca6fb1b7c988d5bb",
  "integration": {
    "name": "PIPEDREAM"
  },
  "integrations": {
    "PIPEDREAM": false
  },
  "type": "track",
  "properties": {
    "artist": "Artist One",
    "genre": "Jazz",
    "song": "Take Five"
  }
}
```

<br />

> ### Event transformation

Reviewer 2: Should we move **## Event transformation** above **## How RudderStack creates the event payload** similar to the other source docs? It gives a little more clarity by going through the mappings first and then seeing the payload. WDYT?

Author: Agreed! Made the change.
